sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -212,7 +214,12 @@ def _batch_send(
         try:
             if response.status_code not in [202]:
                 raise HttpResponseError(response=response)
-            return response.parts()
+            parts = response.parts()
+            if raise_on_any_failure:
+                failures = [p for p in parts if p.status_code not in [202]]
```

Comment:
Just a thought - couldn't we break out of the loop early after finding one failure?

Reply:
no...we want all the failures to be listed

Reply:
Will the success status code for each individual part _also_ always be 202?

Reply:
changed it to [200, 202]

Reply:
Actually we can do: not 200 <= p.status_code < 300 that will cover all successful cases